src/lib.rs

```diff
@@ -100,48 +102,21 @@ fn args_are_cpp(clang_args: &[String]) -> bool {
         .any(|w| w[0] == "-x=c++" || w[1] == "-x=c++" || w == &["-x", "c++"]);
 }
 
-/// A type used to indicate which kind of items do we have to generate.
-///
-/// TODO(emilio): Use `bitflags!`
-#[derive(Debug, Clone)]
-pub struct CodegenConfig {
-    /// Whether to generate functions.
-    pub functions: bool,
-    /// Whether to generate types.
-    pub types: bool,
-    /// Whether to generate constants.
-    pub vars: bool,
-    /// Whether to generate methods.
-    pub methods: bool,
-    /// Whether to generate constructors.
-    pub constructors: bool,
-    /// Whether to generate destructors.
-    pub destructors: bool,
-}
-
-impl CodegenConfig {
-    /// Generate all kinds of items.
-    pub fn all() -> Self {
-        CodegenConfig {
-            functions: true,
-            types: true,
-            vars: true,
-            methods: true,
-            constructors: true,
-            destructors: true,
-        }
-    }
-
-    /// Generate nothing.
-    pub fn nothing() -> Self {
-        CodegenConfig {
-            functions: false,
-            types: false,
-            vars: false,
-            methods: false,
-            constructors: false,
-            destructors: false,
-        }
+bitflags! {
+    /// A type used to indicate which kind of items we have to generate.
+    pub struct CodegenConfig: u32 {
+        /// Whether to generate functions.
+        const FUNCTIONS = 0b1 << 0;
+        /// Whether to generate types.
+        const TYPES = 0b1 << 1;
+        /// Whether to generate constants.
+        const VARS = 0b1 << 2;
+        /// Whether to generate methods.
+        const METHODS = 0b1 << 3;
+        /// Whether to generate constructors
+        const CONSTRUCTORS = 0b1 << 4;
+        /// Whether to generate destructors.
+        const DESTRUCTORS = 0b1 << 5;
     }
```

Comment:
It may be worth to keep convenience methods like:

```rust
pub fn types(self) -> bool {
    self.contains(CodegenConfig::TYPES)
}
```

And such, so the conditionals below and in `codegen.rs` read a bit nicer. Though not strong opposition if you don't feel like doing it :)

Reply:
Added, the code reads better with them.